### PR TITLE
Standardize implementation to look like other functions

### PR DESCRIPTION
**Pull request by gotwarlost**
* The function is now implemented in the root package as opposed to a `server` subcommand. Uses Kong for CLI parsing
* Tooling is moved to a new command called `fn-cue-tools`
* Function inputs are simplified to lose the intermediate `spec` attribute which other functions do not have and have the standard package layout for inputs (the `input` subdirectory as opposed to `pkg/input`)
* Elastic-specific files like documenting dependencies, notices and scripts for generating them are removed
* Examples are fixed to use the latest tooling code. It still has a dependency on a test image I pushed to my docker hub account. This can only be fixed once we have an image pushed to the right place. Will handle that in a separate PR.

### CODE REVIEW

**Comment by gotwarlost** [comment]
Still need to work out what, if anything, needs to happen to copyright clauses etc. Will handle that in a separate PR.

File: internal/fn/fn.go [hunk 1]
```diff
@@ -21,25 +21,23 @@ import (
...
 	"github.com/crossplane/function-sdk-go/response"
-	input "github.com/elastic/crossplane-function-cue/pkg/input/v1beta1"
 	"github.com/pkg/errors"
 	"google.golang.org/protobuf/encoding/protojson"
-	"google.golang.org/protobuf/types/known/durationpb"
 	"google.golang.org/protobuf/types/known/structpb"
 )
 
-const debugAnnotation = "crossplane-function-cue/debug"
+const debugAnnotation = "function-cue/debug"
```
Comment (negz) [inline internal/fn/fn.go]: Maybe `cue.fn.crossplane.io/debug`? We often use crossplane.io in annotation keys.

**Review by negz — APPROVED** [review]
Thanks! All the typical function patterns look good to me in this PR.

File: internal/fn/fn.go (hunk 1 above)
Comment (gotwarlost) [reply internal/fn/fn.go]: Sounds good. I'll make this change. Wasn't sure what namespace to use.

Comment (gotwarlost) [reply internal/fn/fn.go]: Fixed in https://github.com/crossplane-contrib/crossplane-function-cue/pull/1/commits/35341ddffcc33886aff8c5adb381f736a5724233

**Pull request merged**